Update README

ewynx opened the pull request:
# Description

I updated the README for this library. Was planning to only fill out some TODOs, but found myself giving it a bit of a makeover, with the aim to help new devs to this library to get started. 

Hopefully this is helpful. Please let me know if things have to be changed!

## Problem\*

Resolves <!-- Link to GitHub Issue -->

## Summary\*



## Additional Context



# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

jzaki commented on README.md:
```diff
@@ -38,203 +30,404 @@ Add imports at the top of your Noir code, for example:
+This library includes predefined parameter sets. Refer to below for preset info and how to customize the `Params` correctly. 
 
-### Usage
+The actual value of a BigNum can be calculated by multiplying each limb by an increasing power of $2^{120}$. For example `[1,20,300]` represents $1*2^{120*0} + 20*2^{120*1} + 300*2^{120*2}$. We say that the BigNum is represented in radix-$2^{120}$. 
```
The `*` does not render, consider `\cdot` for the dot operator.

ewynx replied:
Fixed this with \cdot and an additional space after "radix-"

jzaki reviewed (dismissed):
I'm happy with the README.md updates, and understand at a high level the U1024 change.

ewynx commented:
Added an examples folder with 3 different projects using: a preset, a custom compile-time modulus and a run-time modulus. 

To have those examples run well, the `src` folder and the main `Nargo.toml` should ideally be placed in a separate folder (for example `lib`). Is that a reorganisation you're ok with?

ewynx commented:
PR is ready for re-review. Moved the original bignum code to folder lib, now the examples will be able to execute correctly. Also adjusted the bignum dependency for the examples to be the local folder, in line with the strategy of [this PR](https://github.com/noir-lang/noir_rsa/pull/4).
